scene/3d/skeleton_3d.cpp

```diff
@@ -865,7 +865,7 @@ void Skeleton3D::set_bone_pose_rotation(int p_bone, const Quaternion &p_rotation
 	const int bone_size = bones.size();
 	ERR_FAIL_INDEX(p_bone, bone_size);
 
-	bones[p_bone].pose_rotation = p_rotation;
+	bones[p_bone].pose_rotation = p_rotation.normalized();
```

Comment:
```suggestion
	ERR_FAIL_COND_EDMSG(!p_rotation.is_normalized(), "The quaternion bone " + get_bone_name(p_bone) + " pose rotation must be normalized.");
	bones[p_bone].pose_rotation = p_rotation;
```

Reply:
- Although point 1. is a genuine concern what you point out in 2. is a "problem" in all quaternions in the engine as showcased here :

![cube](https://github.com/user-attachments/assets/3e3128a1-e830-4f6e-9d80-fab02c5d41b5)

- Other than that the suggested change although great in theory in practice it leaves the editor cluttered with error messages upon any change to the quaternion.

![norm](https://github.com/user-attachments/assets/d2edf059-cf3e-4632-857f-010a91cf893a)

the user will have to use a calculator each time they want to change rotation, aside from that even when a normalized quaternion is used (showcased in the last gif when dragging Temporary Euler, because the quaternions they outputs are always normalized) the error still shows up presumably because of a precision issue. 
if we are to go with the error approach it would definitely require more tweeking.

- That's why I think what you brought up in point 3. and in the end is what should be implemented, a 3rd editor for non-normalized quaternions similar to Temporary Euler but then this issue becomes related to UI and UX too which complicates things further.

- But in the end this issue seems to be more about how the entire engine handles Quaternions and is not exclusive to skeleton3D so we might need more opinions on this.

Reply:
> Other than that the suggested change although great in theory in practice it leaves the editor cluttered with error messages upon any change to the quaternion.

Yes, so I mentioned that a temporary Quaternion PR was needed first. In any case, the conclusion is that normalization should not be done this line.